### DISCUSSION TITLE
Neopixel Problem with Background

### DISCUSSION THREAD

**Discussion by Kamseder** [mriscoc/Ender3V2S1#1110] [General]
Hello There, i don´t know how to Report a bug, but i guess i fund one or maybe someone could help me figure it out.

First: my language is not native English so sorry for that
Second: I am no Programmer but a buddy who is a professional helped me (with no result)

i used the lastest Release (September) and Build a Custom Firmware with:

Printer: Ender3V2
Board: 422
Leveling: BLT/UBL
Display: DWIN
Termistor: (T5 Because i Upgraded to E3D v6)
Features: IS, LA, MPC, NP (NP Cause of Stealthburner toolhead with 3 Neopixels)

1. Put everything togehter and played around with the Neopixels
2. Wanted the Voronlogo Orange and the Hotend ones White
3. Set Orange up in the Configuration_adv.h under the LED_Color_Presets
4. Set White in the NEOPIXEL_BKGD_ stuff and there the fun beginns,

Everytime i uncomment something in the NEOPIXEL_BKGD after flashing the firmware the rotary encoder is super super slow (only we could guess is cause of the stress for the cpu?!)

I can only set the NEOPIXEL_BKGD_TIMEOUT_COLOR, if i wanna set the  NEOPIXEL_BKGD_ALWAYS_ON (for lightning the hotend) it says it wan´t compile cause of indifferences with the LCD_BACKLIGHT_TIMEOUT (in Configuration_adv)

But i can´t coment the  LCD_BACKLIGHT_TIMEOUT with // beacause it gives me a strange error (whats sad because in the old versions it was always on and the timeout was disabled)

So our guess was that the rotary encoder was not manly slow because of the Neopixel Background Stuff, but mainly because it puts to much stress on the cpu cause of not be able to dissable the LCD Backlight Timeout, so that may the CPU has to Count for the Display and for the 2 Neopixels is to much.

Would be nice if sombody could have a look on it, mainly of the LCD Backlight Timeout which i guess is clearly a bug... and maaaaaayy why the Neopixel Background Stuff slowes down the rotary encoder that much that it is unusable.

**Comment by mriscoc** [comment]
LCD Backlight Timeout can be disabled with [M255 S0](https://marlinfw.org/docs/gcode/M255.html)

**Reply by Kamseder** [reply]
first part done :) now the question is why the rotary encoder gets slow even when the timeout is dissabled and i uncomment something from the NEOPIXEL_BKGD part
